src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.sol

```diff
@@ -2,13 +2,66 @@ pragma solidity ^0.4.11;
 
 contract CertificateAuthorityManager {
 
-	event CAQueriedFor(address sender, uint slotNumber);
+	address[] internal certificateAuthorities;
 
-	address a1 = 0x00edde8656c35fcb7126c61fc6e2673734425a72bf;
+    function CertificateAuthorityManager(address initialCertificateAuthority) public {
+        certificateAuthorities = [initialCertificateAuthority];
+    }
 
-	function isCertificateAuthorityFor(address sender) constant returns(bool) {
-		CAQueriedFor(sender, 0);
-		// TODO: Implement.
-		return a1 == sender;
+	function getCertificateAuthorities() public constant returns(address[]) {
+		return certificateAuthorities;
 	}
+
+	function isElectedCAForNextBlock(address ca, uint slotNumber) public constant returns(bool) {
+		uint electedCAIndex = slotNumber % certificateAuthorities.length;	
+		return certificateAuthorities[electedCAIndex] == ca;
+	}
+
+	function addCA(address newCA) public returns(address[]) {
```

Comment:
Maybe this function and `removeCA` should return no value as it's not needed?

Reply:
:100: